src/why3prop.ml

```diff
@@ -0,0 +1,122 @@
+    "Why3axiom_" ^ (string_of_int !nbr_axioms)
+
+(* builtins configuration for propositional logic *)
+type prop_config =
+  { symb_P     : sym * pp_hint (** Encoding of propositions.        *)
+  ; symb_T     : sym * pp_hint (** Encoding of types.               *)
+  ; symb_or    : sym * pp_hint (** Disjunction(∨) symbol.           *)
+  ; symb_and   : sym * pp_hint (** Conjunction(∧) symbol.           *)
+  ; symb_imp   : sym * pp_hint (** Implication(⇒) symbol.           *)
+  ; symb_bot   : sym * pp_hint (** Bot(⊥) symbol.                   *)
+  ; symb_not   : sym * pp_hint (** Not(¬) symbol.                   *) }
+
+(** [get_prop_config builtins] set the builtins configuration using
+    [builtins] *)
+let get_prop_config : Proof.builtins -> prop_config = fun builtins ->
+  let find_sym key =
+    try StrMap.find key builtins with Not_found ->
+     Console.fatal_no_pos "Builtin symbol [%s] undefined." key
+  in
+  { symb_P     = find_sym "P"
+  ; symb_T     = find_sym "T"
+  ; symb_or    = find_sym "or"
+  ; symb_and   = find_sym "and"
+  ; symb_imp   = find_sym "imp"
+  ; symb_bot   = find_sym "bot"
+  ; symb_not   = find_sym "not" }
+
+(** [translate builtins (hs, g)] translate from lambdapi to Why3 goal [g]
+    using the hypothesis [hs]. *)
+let rec translate : Proof.builtins -> (Env.env * term) ->
+    cnst_table * (string * Why3.Term.term) list * Why3.Term.term =
+    fun builtins  (hs, g) ->
+    let cfg = get_prop_config builtins in
+    let (l_prop, hypothesis) =
+        List.fold_left (translate_context cfg) ([], []) hs in
+    try
+        let (l_prop, formula) = t_goal cfg l_prop g in
+        (l_prop, hypothesis, formula)
+    with NoGoalTranslation ->
+        Console.fatal_no_pos "The term [%a] is not of the forme [P _]"
+        Print.pp g
+
+(** [translate_context cfg (l_constants, l_hypothesis) (hyp_name, (_, hyp))]
+    translate the context [hyp] with the label [hyp_name] and add it in
+    [l_hypothesis] with the why3 constants [l_constants]. *)
+and translate_context :
+    prop_config ->
+    cnst_table * (string * Why3.Term.term) list ->
+    string * (tvar * tbox) ->
+    cnst_table * (string * Why3.Term.term) list =
+    fun cfg (l_constants, l_hypothesis) (hyp_name, (_, hyp)) ->
+    try
+        let (new_why3_l, hyp') = t_goal cfg l_constants (Bindlib.unbox hyp) in
+            (new_why3_l, (hyp_name, hyp')::l_hypothesis)
+    with NoGoalTranslation ->
+            (l_constants, l_hypothesis)
+
+(** [t_goal cfg l_prop trm] translate the lambdapi term [trm] to Why3 term
+    using the configuration [cfg] and the list of Why3 constants in [l_prop].
+    *)
+and t_goal : prop_config -> cnst_table -> term ->
+    cnst_table * Why3.Term.term =
+    fun cfg l_prop trm ->
+    match Basics.get_args trm with
+    | (symbol, [t]) when Basics.is_symb (fst cfg.symb_P) symbol ->
+        t_prop cfg l_prop [] t
+    | _                                                         ->
+        raise NoGoalTranslation
```

Comment:
Is `NoGoalTranslation` really useful? Why not always returning a fresh proposition symbol when you don't know how to translate something like you do in t_prop?

Reply:
A => (A or B) is provable even if B is not...

Reply:
The `=>` isn't matched here then, all the expression (`A => (A or B)`) will be translated as a fresh proposition. It should be fixed when we add the handling of `Prod`.
The `or` here is an operator that takes two `Prop` or two proofs of `Prop`?

Reply:
If the context is inconsistent, then a proof could be found even if the goal is abstracted. It may not happen very often thought. Should probably print a warning if the goal is completely abstracted.

Reply:
If the context is inconsistent, then a proof could be found even if the goal is abstracted. It may not happen very often thought. Should probably print a warning if the goal is completely abstracted.

Reply:
If the context is inconsistent, then a proof could be found even if the goal is abstracted. It may not happen very often thought. Should probably print a warning if the goal is completely abstracted.

Reply:
If the context is inconsistent, then a proof could be found even if the goal is abstracted. It may not happen very often thought. Should probably print a warning if the goal is completely abstracted.

Reply:
As already discussed, thanks to add some code for handling products.

Reply:
And thanks to add some doc in doc/sections/tactics.md.